reference/docs-conceptual/SecretManagement/how-to/using-secrets-in-automation.md

```diff
@@ -0,0 +1,84 @@
+---
+description: This article explains how to use a SecretStore vault in an automation scenario to securely retrieve an use passwords or other secret material.
+ms.date: 06/24/2022
+title: Using the SecretStore in automation
+---
+# Using the SecretStore in automation
+
+This article provides an example for using a **Microsoft.PowerShell.SecretStore** vault in an
+automation scenario. A **SecretStore** vault provides you a way to securely store and retrieve the
+passwords, tokens and other secrets you need to use in your automation pipeline.
+
+## Setting up the host the runs the automation
+
+For this example you must first install and configure the SecretManagement modules. This example
+assume that your automation host is running Windows. These commands must be run in the user context
+of the automation account on the host.
+
+```powershell
+Install-Module -Name Microsoft.PowerShell.SecretStore -Repository PSGallery -Force
+Install-Module -Name Microsoft.PowerShell.SecretManagement -Repository PSGallery -Force
+Import-Module Microsoft.PowerShell.SecretStore
+Import-Module Microsoft.PowerShell.SecretManagement
+```
+
+You must also create a password as a **SecureString** that is securely exported to an XML file and
+encrypted by Windows Data Protection (DPAPI). The following command prompts you for a password. In
+this example the **UserName** is unimportant.
+
+```powershell
+PS> $credential = Get-Credential -UserName 'SecureStore'
+
+PowerShell credential request
+Enter your credentials.
+Password for user SecureStore: **************
+```
+
+Once you have the password you can save it to an encrypted XML file.
+
+```powershell
+$securePasswordPath = 'C:\automation\passwd.xml'
+$credential.Password |  Export-Clixml -Path $securePasswordPath
+```
+
+Next you must configure the **SecretStore** vault. The configuration sets user interaction to
```

Comment:
A user might ask 'Why not just use SecureString and Export/Import-Clixml for all my secret needs?'.  The answer is that SecretManagement is more convenient in that it supports secret types other than string types, and provides a uniform method to set/retrieve secrets.